### PR TITLE
swupdate: run as a service

### PR DESCRIPTION
**Pull request by TSAnsari**
Signed-off-by: Tariq Ansari <tansari@luxoft.com>

### CODE REVIEW

**Comment by sashko** [comment]
Did you write or copied those lua scripts from somewhere?

**Comment by TSAnsari** [comment]
The scripts are from before. They had preinst function implemented. I have only added postinst function to them. Also split the single emmcsetup.lua scripts into 2 emmcsetup1.lua and emmcsetup2.lua as logic in postinst changes for different partitions. 

And there was no way of creating a helper lua script that contained common functionality, hence, this duplication of code.

File: recipes-extended/images/pelux-image-update/intel-corei7-64/emmcsetup1.lua [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+function file_exists(name)
+	local f=io.open(name,"r")
+	if f~=nil then io.close(f) return true else return false end
+end
+
+function cmdexec(cmd)
+	local ret, s, status = os.execute(cmd)
+	if (status ~= 0) then
+		return false, cmd .. " return with error"
```
Comment (JEderonn) [inline recipes-extended/images/pelux-image-update/intel-corei7-64/emmcsetup1.lua]: would be good to indicate the value of status in the error message

File: recipes-extended/images/pelux-image-update/arp/emmcsetup1.lua [hunk 2]
```diff
@@ -81,5 +81,35 @@ end
 function postinst()
 	local out = "Post installed script called"
 
+	ret, out = cmdexec("mkdir -p /tmp/mountedsda2")
+	if (ret == false) then
```
Comment (JEderonn) [inline recipes-extended/images/pelux-image-update/arp/emmcsetup1.lua]: A general comment. It's good practice to do comparisons such as
IF (CONSTANT == VARIABLE)
i.e.
if (false == ret)
if (1 == j)
etc

Yes, good compilers will find and notify about potential errors like:
if (ret = false)
but in some cases the code will actually be easier/faster to read if the variable is put at the beginning.

File: recipes-extended/images/pelux-image-update/intel-corei7-64/emmcsetup1.lua [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	if (found) then
+		f:close()
+		fo:close()
+		return true, out
+	end
+
+	start=start+size
+	partitions = eMMC .. "3 : start=    " .. string.format("%d", start) .. ", size=  " .. size .. ", type=83\n"
+
+	fo:write(partitions)
```
Comment (JEderonn) [inline recipes-extended/images/pelux-image-update/intel-corei7-64/emmcsetup1.lua]: what if an error occur when writing partitions?

File: recipes-extended/images/pelux-image-update/intel-corei7-64/emmcsetup1.lua [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+	fo:close()
+	f:close()
+
+	out = os.capture("/usr/sbin/sfdisk --force " .. eMMC .. " < /tmp/partitions")
```
Comment (JEderonn) [inline recipes-extended/images/pelux-image-update/intel-corei7-64/emmcsetup1.lua]: handle any error

File: recipes-extended/images/pelux-image-update/intel-corei7-64/emmcsetup1.lua [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+	-- use partprobe to inform the kernel of the new partitions
+
+	cmdexec("/usr/sbin/partprobe " .. eMMC)
```
Comment (JEderonn) [inline recipes-extended/images/pelux-image-update/intel-corei7-64/emmcsetup1.lua]: what if the command fails? handle error

**Comment by TSAnsari** [comment]
> There are a lot of duplicate code (lua-code). Would it be possible to refactor the code/files, and put the duplicate code into a file with common functions taking the variants as arguments?

Yes I tried to find a way to refactor code to so that there is no duplication. But its not possible to add a common helper lua file based on the swupdate design. We can only add one lua file per handler. So everything needs to be in that file. I discussed this with swupdate author and his reply was this is not supported.

File: recipes-extended/images/pelux-image-update/arp/emmcsetup1.lua (hunk 2 above)
Comment (TSAnsari) [reply recipes-extended/images/pelux-image-update/arp/emmcsetup1.lua]: sure this will reduce accidental errors. Will fix that.

**Pull request merged**